### ISSUE TITLE
Issues with basic run

### ISSUE THREAD

**Issue by nealbgit** [sleepingdog/IMdb-sql-queries#1]
Hi -

Just a couple of issues that needed resolving when I ran the scripts (on SQL 2016):
Script 1:
name-basics.xml - required nconst to be changed to length 12 after truncation error
title-principals.xml - required principalCast to be changed to length 512 after truncation error

Script 2:
Got these errors (since tconst is not unique in table titleprincipals):
Msg 1505, Level 16, State 1, Line 46
The CREATE UNIQUE INDEX statement terminated because a duplicate key was found for the object name 'movie.titleprincipals' and the index name 'PK_titleprincipals_tconst'. The duplicate key value is (tt0000001).
Msg 1750, Level 16, State 1, Line 46
Could not create constraint or index. See previous errors.
The statement has been terminated.
Msg 547, Level 16, State 0, Line 72
The ALTER TABLE statement conflicted with the FOREIGN KEY constraint "FK_titleprincipals_tconst". The conflict occurred in database "IMDb", table "movie.titlebasics", column 'tconst'.

Thanks.

**Comment by sleepingdog** [comment]
Thanks, I’ll look into these errors and apply what fixes I can.

I see from the IMDb data discussion that massive duplication of IDs was reported a couple of days ago:

"The Title.basics file today contains 2,232,860 duplicate ID’s”
https://getsatisfaction.com/imdb/topics/imdb-data-now-available-in-amazon-s3?topic-reply-list[settings][filter_by]=all&topic-reply-list[settings][page]=3#topic-reply-list

**Comment by sleepingdog** [comment]
Hi again Neal

Yes, there were two separate issues.

One is that the id columns (nconst and tconst) are no longer to be considered fixed width at 9 characters, so I followed your recommendation and changed them all to have a maximum of 12 characters, which should future-proof it a bit.

The second more complex issue was the (welcome) new, improved structure to the title.principals table, which normalises the rows into title-person pairs, adding some new columns. I am still not entirely sure about the data integrity, but I have refactored the relevant code and hopefully now it will work with the latest .tsv files.

My comment about title.basics duplicates turned out to be a red herring, it contains alternate titles so that’s why the rows have duplicate ids.

I guess I can close this issue now. Thanks again for noting the problems (I should have fixed them sooner, but got distracted by coursework).

Tavis Reddick
SleepingDog
